elastic/kibana
# [Fleet] added filters to tags query

## Summary

Fixes https://github.com/elastic/kibana/issues/137055

Passing `showInactive` and `kuery` to tags API that populates agent list `Tags` filter.

To verify:
- enroll agents, add different tags to them
- force unenroll one of the agents
- verify that the `Tags` list doesn't show the tag of inactive agent
- change filter to `Status: Inactive` and verify that `Tags` list shows the tag of inactive agent
- similarly, Tags list should change when you select other filters or enter a query.


https://user-images.githubusercontent.com/90178898/180801646-d0c6a109-b48d-45e7-a842-e75118b17736.mov


Note: decided to leave out `showUpgradeable` flag, as the [logic](https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/server/services/agents/crud.ts#L208-L227) of determining upgradeable can't be done on ES side, so would require querying agents and filter in memory, which seems like an overkill for filtering tags.


### Checklist
